components/ui/form/Schedule.tsx

```diff
@@ -62,6 +63,36 @@ const TimeRangeField = ({ name }: TimeRangeFieldProps) => {
     );
   }, []);
 
+  const [valueSelectStart, setValueSelectStart] = useState<Option>(initialValueforSelectStart);
+
+  const createDateAndModifyNewValue = (onChange: any, option: any) => {
+    onChange(new Date(option?.value as number));
+    setValueSelectStart(option);
+  };
+
+  const isInTimeRange = (value: Option) => {
+    const earlyMorningHours = [{ open: "00:00", close: "04:45" }];
+
+    earlyMorningHours.forEach((item) => {
+      const open = new Date("1/1/1999 " + item.open);
+      const close = new Date("1/1/1999 " + item.close);
+      const hourDate = new Date(`1/1/1999 ${value.label}`);
+      if (hourDate >= open && close >= hourDate) {
+        return true;
+      }
```

Comment:
Returning from the `forEach` callback does nothing to the outer scope. You can refactor it using a classic for, or check the `some` method!